iothub/device/src/IotHubBaseClient.cs

```diff
@@ -62,9 +62,10 @@ public abstract class IotHubBaseClient : IAsyncDisposable
                 DesiredPropertyUpdateCallback = OnDesiredStatePatchReceived,
                 ConnectionStatusChangeHandler = OnConnectionStatusChanged,
                 MessageEventCallback = OnMessageReceivedAsync,
+                RetryPolicy = _clientOptions.RetryPolicy ?? new IotHubClientNoRetry(),
```

Comment:
Set RetryPolicy with the default exponential backoff policy unless user set it to null which is IotHubClientNoRetry()